bodhi/server/validators.py

```diff
@@ -261,11 +288,26 @@ def validate_acls(request):
 
             buildinfo = request.buildinfo[build]
 
-            # Get the RpmPackage object
+            # Figure out what kind of package this should be
+            try:
+                package_class = infer_content_class(
+                    base=Package, build=buildinfo['info'])
+            except Exception as e:
+                error = 'Unable to infer content_type.  %r' % e.message
+                log.exception(error)
+                request.errors.add('body', 'builds', error)
+                if isinstance(e, NotImplementedError):
+                    request.errors.status = HTTPNotImplemented.code
+                return
+
+            # Get the Package object
             package_name = buildinfo['nvr'][0]
-            package = db.query(RpmPackage).filter_by(name=package_name).first()
+            package = db.query(package_class)\
+                .filter_by(name=package_name).first()
```

Comment:
FWIW, it is now possible to do ```package_class.query.filter_by()```, thanks to some fancy pants work by @jeremycline ☺